src/hooks/users.ts

```diff
@@ -1,63 +1,185 @@
 import { useMemo } from "react";
-import { isLoaded } from "react-redux-firebase";
 
 import { User } from "types/User";
-import { ValidStoreAsKeys } from "types/Firestore";
 
-import { usersSelector, usersByIdSelector } from "utils/selectors";
+import {
+  worldUsersSelector,
+  worldUsersByIdSelector,
+  sovereignVenueIdSelector,
+} from "utils/selectors";
 import { WithId } from "utils/id";
+import { normalizeTimestampToMilliseconds } from "utils/time";
 
-import { useSelector } from "hooks/useSelector";
-import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
-import { useVenueId } from "hooks/useVenueId";
-import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { fetchSovereignVenueId } from "api/sovereignVenueId";
 
-const useConnectVenueUsers = () => {
+import {
+  setSovereignVenueId,
+  setSovereignVenueIdIsLoading,
+} from "store/actions/SovereignVenueId";
+
+import { useSelector } from "./useSelector";
+import { useUserLastSeenThreshold } from "./useUserLastSeenThreshold";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+import { useVenueId } from "./useVenueId";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useDispatch } from "./useDispatch";
+
+export const useSovereignVenueId = (venueId?: string) => {
```

Comment:
The name of this entity is up for discussion. But I would not spend much of our mental effort, since it is here not for long(until we implement world entities)

Reply:
- https://www.dictionary.com/browse/sovereign
  - > having supreme rank, power, or authority
  
Seems a valid name to imply it's purpose.
  
For searchability/explicitness, this is the concept that we have referred to in other places in the past as 'root parent venue' and similar.